### ISSUE TITLE
"tls-rustls" couldn't compile

### ISSUE THREAD

**Issue by Nathan-M-code** [UkoeHB/bevy_simplenet#4]
Hello, I am facing a compilation error on stable-x86_64-unknown-linux-gnu, with "tls-rustls" feature.
Also I was wondering when the Authenticator::Token feature will be implemented ?

```
   Compiling bevy_simplenet v0.12.0
error[E0308]: `match` arms have incompatible types
  --> /home/me/.cargo/registry/src/index.crates.io-6f17d22bba15001f/bevy_simplenet-0.12.0/src/server/server.rs:75:44
   |
71 |       let server = match acceptor_config
   |  __________________-
72 | |     {
73 | |         AcceptorConfig::Default         => server.acceptor(axum_server::accept::DefaultAcceptor::new()),
   | |                                            ------------------------------------------------------------ this is found to be of type `axum_server::Server`
74 | |         #[cfg(feature = "tls-rustls")]
75 | |         AcceptorConfig::Rustls(config)  => server.acceptor(axum_server::tls_rustls::RustlsAcceptor::new(config...
   | |                                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `Server`, found `Server<RustlsAcceptor>`
76 | |         #[cfg(feature = "tls-openssl")]
77 | |         AcceptorConfig::OpenSSL(config) => server.acceptor(axum_server::tls_openssl::OpenSSLAcceptor::new(config...
78 | |     };
   | |_____- `match` arms have incompatible types
   |
   = note: expected struct `axum_server::Server<DefaultAcceptor>`
              found struct `axum_server::Server<RustlsAcceptor>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `bevy_simplenet` (lib) due to 1 previous error
```

Thank you !

**Comment by UkoeHB** [comment]
> Hello, I am facing a compilation error on stable-x86_64-unknown-linux-gnu, with "tls-rustls" feature.

Fixed in v0.12.1: https://github.com/UkoeHB/bevy_simplenet/commit/b0abbb89130b31114a13bdb3b753f73ab178f424

> Also I was wondering when the Authenticator::Token feature will be implemented ?

I can work on that this week. You are the first person to ask about it :)

**Comment by UkoeHB** [comment]
@Nathan-M-code `AuthToken` is implemented in v0.13.0: https://github.com/UkoeHB/bevy_simplenet/commit/e864427e339c28aa55cc47050242e55f4ee529a5. I added a section to the README about how to use it.
